File: Tcf.vb
Imports System

Public Class Test
    Public Shared Sub Main()
        Try
            Try
                Throw New Exception("Foo")
            Catch ex as Exception
                Console.WriteLine("Inner Catch")
                Throw
            Finally
                Console.WriteLine("Inner Finally")
            End Try
        Catch ex as Exception
            Console.WriteLine("Outer Catch")
        Finally
            Console.WriteLine("Outer Finally")
        End Try
    End Sub
End Class

File: Tcf.cs
using System;

public class Test
{
	public static void Main()
	{
		try {
			try {
				throw new Exception("Foo");
			}
			catch (Exception ex) {
				Console.WriteLine("Inner Catch");
				throw ex;
			}
			finally {
				Console.WriteLine("Inner Finally");
			}
		}
		catch (Exception ex) {
			Console.WriteLine("Outer Catch");
		}
		finally {
			Console.WriteLine("Outer Finally");
		}
	}
}
